CondFormats/JetMETObjects/src/JetCorrectorParametersHelper.cc

```diff
@@ -0,0 +1,163 @@
+          mBinBoundaries[j].push_back(mRecordsLocal[i].xMin(j));
+        else if(j==SIZE-1) {
+          if(i==0)
+            mBinBoundaries[j].reserve(mRecordsLocal.size());
+
+          mBinBoundaries[j].push_back(mRecordsLocal[i].xMin(j));
+
+          if(SIZE>1 && (i==nRec-1 || mRecordsLocal[i].xMin(j-1)!=mRecordsLocal[i+1].xMin(j-1))) {
+            end = i;
+            //mMap.emplace(gen_tuple<SIZE-1>([&](size_t k){return mRecordsLocal[i].xMin(k);}),std::make_pair(start,end));
+            mMap.emplace(gen_tuple<JetCorrectorParameters::MAX_SIZE_DIMENSIONALITY-1>([&](size_t k){return (k<SIZE-1)?mRecordsLocal[i].xMin(k):-9999;}),std::make_pair(start,end));
+            start = i+1;
+          }
+        }
+      }
+      indexMapSize = mIndexMap.size();
+      tuple_type tmpTuple = gen_tuple<JetCorrectorParameters::MAX_SIZE_DIMENSIONALITY>([&](size_t k){return (k<SIZE)?mRecordsLocal[i].xMin(k):-9999;});
+      mIndexMap.emplace(tmpTuple,i);
+      tmpIndexMapSize = mIndexMap.size();
+      if(indexMapSize==tmpIndexMapSize)
+      {
+        size_t existing_index = mIndexMap.find(tmpTuple)->second;
+        std::stringstream sserr;
+        sserr<<"Duplicate binning in record found (existing index,current index)=("
+             <<existing_index<<","<<i<<")"<<std::endl<<"\tBins(lower bounds)="<<tmpTuple;
+        handleError("JetCorrectorParametersHelper",sserr.str());
+      }
+    }
+  if (mBinBoundaries[SIZE-1].size()!=nRec)
+    {
+      std::stringstream sserr;
+      sserr<<"Did not find all bin boundaries for dimension "<<SIZE-1<<"!!!"<<std::endl
+           <<"Found "<<mBinBoundaries[SIZE-1].size()<<" out of "<<nRec<<" records";
+      handleError("JetCorrectorParametersHelper",sserr.str());
+    }
+  indexMapSize = mIndexMap.size();
+  if (indexMapSize!=nRec)
+    {
+      handleError("JetCorrectorParametersHelper","The mapping of bin lower bounds to indices does not contain all possible entries!!!");
+    }
+}
+//------------------------------------------------------------------------
+//--- JetCorrectorParameters::JetCorrectorParametersHelper sanity checks -
+//--- checks that some conditions are met before finding the bin index ---
+//------------------------------------------------------------------------
+void JetCorrectorParametersHelper::binIndexChecks(unsigned N, const std::vector<float>& fX) const
+{
+  if (N != fX.size())
+    {
+      std::stringstream sserr;
+      sserr<<"# bin variables "<<N<<" doesn't correspont to requested #: "<<fX.size();
+      handleError("JetCorrectorParametersHelper",sserr.str());
+    }
+}
+bool JetCorrectorParametersHelper::binBoundChecks(unsigned dim, const float& value, const float& min, const float& max) const
+{
+  if (value < min || value > max)
+    {
+      std::stringstream sserr;
+      sserr<<"Value for dimension "<<dim<<" is outside of the bin boundaries"<<std::endl
```

Comment:
Hi @aperloff - just make a new pull request that includes this one + your fix. Thanks! (4MB/event of printout is a bit more than "annoying"...)